android/titanium/src/java/org/appcelerator/titanium/TiBaseActivity.java

```diff
@@ -760,6 +762,34 @@ public void onChanged(TiActivitySafeAreaMonitor monitor)
 					windowProxy.fireSafeAreaChangedEvent();
 				}
 			}
+
+			@Override
+			public void onKeyboardChanged(boolean isVisible, int width, int height, Insets keyboardSize)
+			{
+				if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q
+					&& isVisible != keyboardVisible
+					&& tiApp != null
+					&& tiApp.hasListener("keyboardframechanged")) {
```

Comment:
Move the listener-check to the `onApplyWindowInsets` check, so this callback won't even be called if not explicitly listened to.

Reply:
I keep the check still in here. It should only be called when `tiApp != null && tiApp.hasListener("keyboardframechanged")` anyways but for safety I'll check it still here